File: Programming-languages-and-laboratory/Python/Exercises/41_compito_java_20221110/GestionePalestra.py
from Dipendente import Dipendente
from Trainer import Trainer
from Nutrizionista import Nutrizionista
from Cliente import Cliente
from Servizio import Servizio


import sys


class GestionePalestra:
    def main():

        dipendenti = []
        clienti = []
        cod_to_dip = {}

        try:
            f = open("dipendenti.txt", "r")
            line = f.readline().strip()

            while line != "":
                tok = line.split()
                codice = int(tok[0])
                tipo = tok[1]
                nome_dipendente = f.readline().strip()

                if tipo == "trainer":
                    laurea = f.readline().strip()
                    specialita = f.readline().strip()
                    line = f.readline().strip()
                    tok = line.split()
                    ore_settimanali = float(tok[0])
                    costo_orario = float(tok[1])

                    tr = Trainer(codice, tipo, nome_dipendente, laurea, specialita, ore_settimanali, costo_orario)
                    dipendenti.append(tr)
                    cod_to_dip[codice] = tr

                else:
                    line = f.readline().strip()
                    tok = line.split()
                    costo_orario = float(tok[0])
                    telefono = int(tok[1])
                    medico = bool(tok[2])
                    app_sett = int(tok[3])

                    nu = Nutrizionista(codice, tipo, nome_dipendente, costo_orario, telefono, medico, app_sett)
                    dipendenti.append(nu)
                    cod_to_dip[codice] = nu

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("tipo, nome, codice, laurea, specialità, ore settimanali, telefono, medico, app.sett., costo orario")

        for d in dipendenti:
            print(d)
        print()

        try:
            f = open("clienti.txt", "r")
            line = f.readline().strip()

            while line != "":
                codice_cliente = int(line)
                nome_cliente = f.readline().strip()

                cl = Cliente(codice_cliente, nome_cliente)
                clienti.append(cl)

                line = f.readline().strip()
                tok = line.split()
                i = 0

                while i < len(tok):
                    codice_dipendente = int(tok[i])
                    ore_servizio = float(tok[i+1])
                    i += 2
                    ser = Servizio(codice_dipendente, ore_servizio)
                    cl.add_servizio(ser)
                    cl.confronta_ore_servizio(ore_servizio, codice_dipendente)
                    cod_to_dip[codice_dipendente].incrementa_servizi()

                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        for c in clienti:
            print(c)
        print()

        try:
            codice_letto = int(sys.argv[1])

            for dip in dipendenti:
                if codice_letto == dip.get_codice():
                    print("Servizi fatti dal dipendente " + str(codice_letto) + ": "  + str(dip.get_servizi_svolti()))

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)


if __name__ == "__main__":
    GestionePalestra.main()

File: II - Linguaggi di programmazione e laboratorio/Esame/Java/41_compito_java_2022_11_10/GestionePalestra.java
import java.util.*;
import java.io.*;

public class GestionePalestra {
	static List<Dipendente> dipendenti = new LinkedList<Dipendente>();
	static List<Cliente> clienti = new LinkedList<Cliente>();
	static Map<Integer, Dipendente> codice_to_dipendente = new HashMap<Integer, Dipendente>();

	public static void main(String[] args) {

		// Punto 1, memorizzazione dipendenti

		try {
			BufferedReader br = new BufferedReader(new FileReader("dipendenti.txt"));
			String line = br.readLine();

			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int codice = Integer.parseInt(tok.nextToken());
				String tipo = tok.nextToken();
				String nomeDipendente = br.readLine();

				if (tipo.equals("trainer")) {
					boolean laurea = Boolean.parseBoolean(br.readLine());
					String specialita = br.readLine();
					line = br.readLine();
					tok = new StringTokenizer(line);
					float oreSettimanali = Float.parseFloat(tok.nextToken());
					float costoOrario = Float.parseFloat(tok.nextToken());
					Trainer t = new Trainer(codice, tipo, nomeDipendente, laurea, specialita, oreSettimanali,
							costoOrario);
					dipendenti.add(t);
					codice_to_dipendente.put(codice, t);
				} else {
					line = br.readLine();
					tok = new StringTokenizer(line);
					float costoOrario = Float.parseFloat(tok.nextToken());
					int telefono = Integer.parseInt(tok.nextToken());
					boolean medico = Boolean.parseBoolean(tok.nextToken());
					int appuntamentiSettimanali = Integer.parseInt(tok.nextToken());
					Nutrizionista n = new Nutrizionista(codice, tipo, nomeDipendente, costoOrario, telefono, medico,
							appuntamentiSettimanali);
					dipendenti.add(n);
					codice_to_dipendente.put(codice, n);
				}

				line = br.readLine();

			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// Punto 2, memorizzazione clienti

		try {
			BufferedReader br = new BufferedReader(new FileReader("clienti.txt"));
			String line = br.readLine();

			while (line != null) {
				int codiceCliente = Integer.parseInt(line);
				String nomeCliente = br.readLine();

				Cliente c = new Cliente(codiceCliente, nomeCliente);
				clienti.add(c);

				line = br.readLine();
				StringTokenizer tok = new StringTokenizer(line);

				while (line != null && tok.hasMoreTokens()) {
					int codiceDipendente = Integer.parseInt(tok.nextToken());
					float oreServizio = Float.parseFloat(tok.nextToken());

					Dipendente dip = codice_to_dipendente.get(codiceDipendente);

					Servizio s = new Servizio(dip, oreServizio);
					c.addServizio(s);
					dip.incrementaNumeroServiziSvolti();		

				}

				line = br.readLine();
			}

			br.close();

		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}

		// Punto 3, stampa elenco di tutti i dipendenti

		System.out.println(
				"Tipo, nome, codice, laurea, specialità, ore settimanali, telefono, medico, app.sett., costo orario");

		for (Dipendente d : dipendenti)
			System.out.println(d);

		System.out.println(); // spazio

		// Punto 4, stampa elenco clienti con dipendente che ha fornito il servizio più
		// lungo ed elenco servizi utilizzato

		System.out.println("codice, nome, dipendente, servizi");
		for (Cliente c : clienti)
			System.out.println(c);

		System.out.println();

		// Punto 5, legge da riga di comando il codice di un dipendente e ne stampa il
		// numero di servizi svolti

		try {
			int codiceInserito = Integer.parseInt(args[0]);

			for (Dipendente d : dipendenti) {
				if (d.codice == codiceInserito)
					System.out.println("Servizi fatti dal dipendente " + codiceInserito + ": " + d.numeroServiziSvolti);
			}
		} catch (Exception e) {
			System.err.println(e);
		}
	}
}